### PR TITLE
feat: bundle extension using webpack

### PR DESCRIPTION
**Pull request by Michsior14**
As stated in the issue #1, this adds a simple webpack configuration and scripts to bundle the extension.

### CODE REVIEW

**Comment by philnash** [comment]
Hi @Michsior14, thanks for this PR! I've actually been trying to do this myself this evening.

Were you able to package this and install it locally and have it work? I am having trouble with that, with my attempt as well as yours.

I think the issue lies in that the ngrok package includes a binary of ngrok itself and when packaging with webpack that doesn't come at all.

It occurs to me that to bundle the extension may require a postinstall step that runs the ngrok packages postinstall [download script](https://github.com/bubenshchykov/ngrok/blob/master/download.js) which will get the latest version of ngrok on install. What do you think?

**Comment by Michsior14** [comment]
> Hi @Michsior14, thanks for this PR! I've actually been trying to do this myself this evening.
> 
> Were you able to package this and install it locally and have it work? I am having trouble with that, with my attempt as well as yours.
> 
> I think the issue lies in that the ngrok package includes a binary of ngrok itself and when packaging with webpack that doesn't come at all.
> 
> It occurs to me that to bundle the extension may require a postinstall step that runs the ngrok packages postinstall [download script](https://github.com/bubenshchykov/ngrok/blob/master/download.js) which will get the latest version of ngrok on install. What do you think?

Hey!
I've added the missing line in `webpack.config.js`, but I think you are right - it would be better to download the binary directly in `postinstall` so `ngrok` could be bundled as well.

**Comment by philnash** [comment]
Now I can't get postinstalls to work. It's late and I need to sleep!

This is looking good, but we need to make sure the ngrok binary can be downloaded for the right system too. Do you know if a postinstall script works in a VSCode extension?

**Comment by Michsior14** [comment]
> Now I can't get postinstalls to work. It's late and I need to sleep!
> 
> This is looking good, but we need to make sure the ngrok binary can be downloaded for the right system too. Do you know if a postinstall script works in a VSCode extension?

I've tried to download `ngrock` on activation (which currently seems to be the only way), but I have some permission issues. I can't spend more time today on this, maybe you will find sth ;)

**Comment by philnash** [comment]
Yup, I think you're right that that's the way. Will take a look at this later.

Thanks for taking a crack at this!

**Comment by Michsior14** [comment]
@philnash I've updated the PR with working solution for binary download ;) I've also added prettier, so there are no issues with styling the code (you have a bit different IDE settings than me and I didn't wanted to influence the way how you format the code).

**Comment by philnash** [comment]
I pulled the latest on this branch, installed dependencies and tried to build in debug mode to test. It brought my machine to a standstill and I’m trying to restart it now! 

Most of this looks good, but we’ve conflated a bunch of stuff now. Is there a way that you could extract the work for #7 and make a separate PR for that? It’s currently the most important part, as the extension only works for Macs right now, and the other bits can wait (though they are long term important).

**Comment by Michsior14** [comment]
> I pulled the latest on this branch, installed dependencies and tried to build in debug mode to test. It brought my machine to a standstill and I’m trying to restart it now!
> 
> Most of this looks good, but we’ve conflated a bunch of stuff now. Is there a way that you could extract the work for #7 and make a separate PR for that? It’s currently the most important part, as the extension only works for Macs right now, and the other bits can wait (though they are long term important).

That's strange, should work without any issues, but yep sure, here you go: #8 .

**Comment by philnash** [comment]
Thank you! I will have a go with #8 right now and come back to webpack.

File: src/ngrok/index.ts [hunk 1]
```diff
@@ -170,51 +172,51 @@ export const start = async () => {
...
     return window.showErrorMessage(
-      "ngrok is not currently running, please start a tunnel before accessing the dashboard"
+      'ngrok is not currently running, please start a tunnel before accessing the dashboard'
     );
   }
 };
 
 export const downloadBinary = () => {
-  const basePath = join(__dirname, "..", "..", "node_modules", "ngrok", "bin");
+  const basePath = join(__dirname, 'bin');
```
Comment (Michsior14) [inline src/ngrok/index.ts:206]: That's the only real change in the code, everything else is done by prettier.

**Comment by philnash** [comment]
I'm having a look at this now and I am having trouble running the plugin with VSCode debug mode (by pressing F5). The build always seems to hang without starting the Extension Development Host.

Killing the build process does start the host, but stops the task watching for changes.

Are you seeing this? Or have you some sort of workaround for it?

**Comment by Michsior14** [comment]
I've added configurations that allows to debug the extension properly ;)

**Comment by philnash** [comment]
Awesome work, thanks! This is now live in version 1.3.0 on the marketplace.

**Pull request merged**